subcommand/injector/command.go

```diff
@@ -353,47 +358,139 @@ func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.Bundle, client
 			}
 		}
 
-		// If there is a MWC name set, then update the CA bundle.
+		// If there is an MWC name set, then update the CA bundle.
 		if isLeader && c.flagAutoName != "" && len(bundle.CACert) > 0 {
-			// The CA Bundle value must be base64 encoded
-			value := base64.StdEncoding.EncodeToString(bundle.CACert)
-			payload := []byte(fmt.Sprintf(
-				`[{
-					"op": "add",
-					"path": "/webhooks/0/clientConfig/caBundle",
-					"value": %q
-				}]`, value))
-
-			var err error
-			switch adminAPIVersion {
-			// AdmissionregistrationV1 is valid in k8s 1.16+
-			case "v1":
-				_, err = clientset.AdmissionregistrationV1().
-					MutatingWebhookConfigurations().
-					Patch(ctx, c.flagAutoName, types.JSONPatchType, payload,
-						metav1.PatchOptions{})
-			// AdmissionregistrationV1beta1 is present in k8s 1.9 - 1.19
-			case "v1beta1":
-				_, err = clientset.AdmissionregistrationV1beta1().
-					MutatingWebhookConfigurations().
-					Patch(ctx, c.flagAutoName, types.JSONPatchType, payload,
-						metav1.PatchOptions{})
-			default:
-				err = fmt.Errorf("unknown Admissionregistration API version %q", adminAPIVersion)
-			}
+			// Check the current bundles and only update if necessary.
+			currentBundles, err := c.getCurrentCABundles(ctx, clientset, adminAPIVersion)
 			if err != nil {
 				c.UI.Error(fmt.Sprintf(
-					"Error updating MutatingWebhookConfiguration: %s",
+					"Error getting MutatingWebhookConfiguration: %s",
 					err))
 				continue
 			}
+			if len(currentBundles) > 0 && bytes.Equal(currentBundles[0], bundle.CACert) {
```

Comment:
Should this be `!bytes.Equal(currentBundles[0], bundle.CACert)`?

Reply:
Oops. Good catch!